### ISSUE TITLE
Loading a Save With a LiquidTank 

### ISSUE THREAD

**Issue by PhilM-47** [javisar/ONI-Modloader-Mods#20]
I'm not sure if this is a linux specific issue, but I referenced it in the linux compat issue for the loader and you asked me to file it separately. In case it helps you track down the issue, I can place a gas tank and load the save successfully, but not the liquid tank.

Player.log:
https://pastebin.com/knUtmter

For your awareness, the two errors at the top and the other errors about gl_ViewID_OVR and internal error shader are "normal" for ONI on linux, but I haven't seen any actual breakage caused by them.

**Comment by javisar** [comment]
Is not a linux specific issue. I was getting the errors but the liquid tank was working for me.

I've made a change to try to fix it. You can try it using the last version of the dlls in the master branch:

https://github.com/javisar/ONI-Modloader-Mods/blob/master/Mods/LiquidTankMod.dll
https://github.com/javisar/ONI-Modloader-Mods/blob/master/Mods/GasTankMod.dll

**Comment by naemr** [comment]
`[= INFO =] Initializing Modloader v0.4.0
[= INFO =] Path to mods is: D:\Games\SteamLibrary\steamapps\common\OxygenNotIncluded\Mods
[= INFO =] Loading mod assemblies
[= INFO =] Loading GasTankMod, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
[= INFO =] Loading LiquidTankMod, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
[= INFO =] Loading ONI-Common, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
[= INFO =] Applying Harmony patches
[= ERROR =] Patching mod GasTankMod, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null failed!
[= ERROR =] System.ArgumentException: No target method specified for class GasTankMod.KSerialization_Manager_Initialize
  at Harmony.PatchProcessor.ProcessType () [0x00000] in <filename unknown>:0 
  at Harmony.PatchProcessor..ctor (Harmony.HarmonyInstance instance, System.Type type, Harmony.HarmonyMethod attributes) [0x00000] in <filename unknown>:0 
  at Harmony.HarmonyInstance.<PatchAll>b__6_0 (System.Type type) [0x00000] in <filename unknown>:0 
  at Harmony.CollectionExtensions.Do[Type] (IEnumerable`1 sequence, System.Action`1 action) [0x00000] in <filename unknown>:0 
  at Harmony.HarmonyInstance.PatchAll (System.Reflection.Assembly assembly) [0x00000] in <filename unknown>:0 
  at ModLoader.ModLoader.ApplyHarmonyPatches (System.Collections.Generic.List`1 modAssemblies) [0x00000] in <filename unknown>:0 
[= ERROR =] Patching mod LiquidTankMod, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null failed!
[= ERROR =] System.ArgumentException: No target method specified for class LiquidTankMod.KSerialization_Manager_Initialize
  at Harmony.PatchProcessor.ProcessType () [0x00000] in <filename unknown>:0 
  at Harmony.PatchProcessor..ctor (Harmony.HarmonyInstance instance, System.Type type, Harmony.HarmonyMethod attributes) [0x00000] in <filename unknown>:0 
  at Harmony.HarmonyInstance.<PatchAll>b__6_0 (System.Type type) [0x00000] in <filename unknown>:0 
  at Harmony.CollectionExtensions.Do[Type] (IEnumerable`1 sequence, System.Action`1 action) [0x00000] in <filename unknown>:0 
  at Harmony.HarmonyInstance.PatchAll (System.Reflection.Assembly assembly) [0x00000] in <filename unknown>:0 
  at ModLoader.ModLoader.ApplyHarmonyPatches (System.Collections.Generic.List`1 modAssemblies) [0x00000] in <filename unknown>:0 
`

**Comment by naemr** [comment]
The errors in my paste were using your updated files just so you know.

**Comment by naemr** [comment]
Would like to point out, your mod loader is NOT compatible with the Expression Upgrade, which makes your fix unusable now that expression is released.

I will give the same advice here that I have given to mod authors on many other early access games.

Do NOT make mods, loaders, content of any kind, for the game's TESTING branch. its too unstable, and many things done in a testing branch of a game never even make it to release.

All content development should focus on the released branch of the game, this is the branch the widest audience will be using, and trying to develop content in the game's testing branch will only cause you, and your audience undue headache which can ultimately lead to loss of audience.

I give this not in aggravation or anger, I give this advice because I like what you are doing, and want to see it thrive, and I have seen a great many projects in a great many early release games die because the authors got lost in the unstable world of testing branch, and ether got overwhelmed trying to keep up with the changes, or lost there audience because players don't want to play a game that crashes every 5 seconds without mods, then add mods that cause more issue.

**Comment by javisar** [comment]
Yeah, I agree. I've already decided to don't do anything in the testing branch after the headache Expression Upgrade. Thanks for the comment.

**Comment by javisar** [comment]
Could you try with the latest version? I don't have any issues with the save/load anymore.

**Comment by PhilM-47** [comment]
The fluid tank is working for me just fine now, thank you.  Has the gas tank been removed though, or am I going blind and just missing it somewhere?
